### PR TITLE
[language] fix flaky rotation capability test

### PR DESCRIPTION
**Pull request by sblackshear**
This test uses an address as an authentication key, which should fail with abort code 12 after the switch to short addresses in 95ffa7240864df2c201950f41db78ca253503001. However, it appears that functional testing framework only detects the failure intermittently for some reason. This is why the test was flaky. That issue is being investigated/fixed separately.

This PR fixes the failing test by saving the old authentication key and restoring it instead of using the address as the auth key. I also needed to add a procedure for extracting the auth key to `LibraAccount`.

## Motivation

Fixing a flaky test.

## Test Plan

Before this PR, the test in question almost always passed. To confirm that the flakiness was in the functional tests matcher, I added an explicit `// check: EXECUTED` after the first transaction in the bad test. This made the test fail consistently (as expected). I  then fixed the test and left the check in place.

### CODE REVIEW

File: language/ir-testsuite/tests/libra_account/remove_then_replace_withdrawal_capability.mvir [hunk 1]
```diff
@@ -1,27 +1,9 @@
...
-  assert(!LibraAccount.delegated_withdrawal_capability(copy(sender)), 50);
-
-  // extracting the capability should flip the flag
-  cap = LibraAccount.extract_sender_withdrawal_capability();
-  assert(LibraAccount.delegated_withdrawal_capability(copy(sender)), 51);
-
-  // restoring the capability should flip the flag back
-  LibraAccount.restore_withdrawal_capability(move(cap));
-  assert(!LibraAccount.delegated_withdrawal_capability(copy(sender)), 52);
+    abort(77)
```
Comment (runtian-zhou) [inline language/ir-testsuite/tests/libra_account/remove_then_replace_withdrawal_capability.mvir]: Should we just delete this test?

Comment (sblackshear) [reply language/ir-testsuite/tests/libra_account/remove_then_replace_withdrawal_capability.mvir]: Oh, my bad; I messed this up locally. Let me fix it.

File: language/ir-testsuite/tests/libra_account/remove_then_replace_rotation_capability.mvir [hunk 2]
```diff
@@ -19,11 +19,13 @@ main() {
   assert(!LibraAccount.delegated_key_rotation_capability(copy(sender)), 52);
 
   // and the sender should be able to rotate once again
-  LibraAccount.rotate_authentication_key(AddressUtil.address_to_bytes(move(sender)));
```
Comment (sblackshear) [inline language/ir-testsuite/tests/libra_account/remove_then_replace_rotation_capability.mvir]: This is what started failing after 95ffa72.

**Pull request closed without merging**